contracts/staking/StakingPool.sol

```diff
@@ -0,0 +1,517 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+import "@openzeppelin/contracts-upgradeable/security/PausableUpgradeable.sol";
+
+import "../RoleAccessControl.sol";
+import "./StakingPoolLib.sol";
+
+contract StakingPool is
+    Initializable,
```

Comment:
nitl remove `Initializable` from the import as it's included by the other parent contracts,